### ISSUE TITLE
How to get original string data back from TFRecordData

### ISSUE THREAD

**Issue by Hannabrahman** [tensorflow/tensorflow#36739]
I followed TensorFlow guide to save my string data using:
```
def _create_string_feature(values):
    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[values.encode('utf-8')]))
```
I also used ```["tf.string", "FixedLenFeature"]``` as my feature original type, and ```"tf.string"``` as my feature convert type.

However, during my training when I run my session and I create iterators, my string feature for a batch size of 2 (for example: ['food fruit', 'cupcake food' ]) would be like below. The problem is that this list is of size 1, and not 2 (batch_size=2), why instances in one batch are sticked (clubbed) together rather than being splitted?

```
[b'food fruit' b'cupcake food']
```
For my other features which are int or float, they are numpy arrays of shape (batch_size, feature_len) which are fine but not sure why string features are not separated in a single batch?

Any help would be appreciated.

**Comment by Saduf2019** [comment]
@Hannabrahman please provide us with complete stand alone code so we could replicate the issue in our environment

**Comment by Hannabrahman** [comment]
@Saduf2019 thanks for reply.
Actually It's not a part of code. However, this is what I want to do. I have an script which prepare my tfrecord data like following:
```
        def _create_int_feature(values):
            return tf.train.Feature(
                int64_list=tf.train.Int64List(value=list(values)))

        def _create_floats_feature(values):
            return tf.train.Feature(float_list=tf.train.FloatList(value=list(values)))

        def _create_string_feature(values):
            return tf.train.Feature(bytes_list=tf.train.BytesList(value=[values.encode('utf-8')])) # m.encode('utf-8') for m in values

        features = collections.OrderedDict()
        features["x1_ids"] = _create_int_feature(fea["x1_ids"])
        features["x1_len"] = _create_int_feature([fea["x1_len"]])
        features["x1x4_ids"] = _create_int_feature(fea["x1x4_ids"])
        features["x1x4_len"] = _create_int_feature([fea["x1x4_len"]])
        features["x4_m"] = _create_floats_feature(m_fea)
        features["label"] = _create_string_feature(label)
```
Here x1_ids, x1_len, x1x4_ids, x1x4_len, and x4_m are all integers and float. However label is string.
I want to use label as string during my training. I want to do some string matching.
When running my session in tf and creating iterators, my other features format and dimension are numpy array (batch_size, feature_len). However my ```features["label"]``` is a list of size one.
this is how I do this:
```
    datasets = {}
    train_dataset = tx.data.TFRecordData(hparams=config_train.train_hparam)
    datasets['train'] = train_dataset
    dev_dataset = tx.data.TFRecordData(hparams=config_train.dev_hparam)
    datasets['dev'] = dev_dataset
    test_dataset = tx.data.TFRecordData(hparams=config_train.test_hparam)
    datasets['test'] = test_dataset
    iterator = tx.data.FeedableDataIterator(datasets)
    batch = iterator.get_next()
    batch_size = tf.shape(batch['x1x4_ids'])[0]

    with tf.Session(config=session_config) as sess:
        sess.run(tf.global_variables_initializer())
        sess.run(tf.local_variables_initializer())
        sess.run(tf.tables_initializer())

###### SOME CODE TO INITIALIZE MY MODEL ########


        iterator.initialize_dataset(sess)

        if FLAGS.do_train:
            for epoch in range(config_train.max_train_epoch):
                print("Training epoch {}".format(epoch))
                _train_epoch(sess, epoch==0)
```

Here is the first part of my train function:
```
    def _train_epoch(sess, initial=False):
        """Trains on the training set, and evaluates on the dev set
        periodically.
        """
        iterator.restart_dataset(sess, 'train')

        while True:
            try:
                # (1) Get data and yy sample
                fetches_data = {
                    'batch': batch,
                    'batch_size': batch_size,
                }
                feed_dict_data = {
                    iterator.handle: iterator.get_handle(sess, 'train'),
                    tx.global_mode(): tf.estimator.ModeKeys.PREDICT,
                }
                rets_data = sess.run(fetches_data, feed_dict_data)
```

My issue is that the ```rets_data["label"]``` is not a list of length batch_size. As I said in my previous post, for example if label before creating tfrecord is ```['food fruit', 'cupcake food' ]``` (just the first two element for example) after doing tfrecord when I print ```rets_data["label"]``` for one batch it would be like:
```
[b'food fruit' b'cupcake food']
```
why they are not separated?

**Comment by gowthamkpr** [comment]
@Hannabrahman Closing this issue as it is not related to bug/performance, build/install, feature request or docs related issues.

Please post this question in stack overflow as there is a bigger community to respond. Thanks!

**Comment by Hannabrahman** [comment]
@gowthamkpr I have already asked it [here](https://stackoverflow.com/questions/60177218/how-to-get-original-string-data-back-from-tfrecorddata). But received no response. So TensorFlow github doesn't allow asking questions?
